lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx

```diff
@@ -69,6 +85,54 @@ export default function AssignmentActivity() {
           {assignment.data && title}
         </h2>
       </div>
+      <DashboardActivityFilters
+        selectedCourseIds={courseId ? [courseId] : []}
+        onCoursesChange={newCourseIds => {
+          // When no courses are selected (which happens if either "All courses" is
+          // selected or the active course is deselected), navigate to "All courses"
+          // section and propagate the rest of the filters.
+          if (newCourseIds.length === 0) {
+            navigate(
+              search
+                ? `?${search}&assignment_id=${assignmentId}`
+                : `?assignment_id=${assignmentId}`,
+            );
+          }
+
+          // When a course other than the "active" one (the one represented
+          // in the URL) is selected, navigate to that course and propagate
+          // the rest of the filters.
+          const firstDifferentCourse = newCourseIds.find(c => c !== courseId);
+          if (firstDifferentCourse) {
+            navigate(
+              `${courseURL(firstDifferentCourse)}?${search}&assignment_id=${assignmentId}`,
+            );
+          }
+        }}
+        selectedAssignmentIds={[assignmentId]}
+        onAssignmentsChange={newAssignmentIds => {
+          // When no assignments are selected (which happens if either "All
+          // assignments" is selected or the active assignment is deselected),
+          // navigate to "The assignment's course" section and propagate the
+          // rest of the filters.
+          if (newAssignmentIds.length === 0 && courseId) {
```

Comment:
In this view there _should_ always be a course, since all assignments exist within a course. If the user changes the course, we'll transition out of this view. It looks to me like the only exception is when loading assignment details and we don't have the course ID yet. In that state we should make it impossible to change the assignment selection.